contracts/src/v0.8/ccip/test/pools/GHO/fork/GhoTokenPoolMigrate1_4To1_5/ForkBase.t.sol

```diff
@@ -0,0 +1,445 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.19;
+
+import {Test} from "forge-std/Test.sol";
+import {IERC20} from "../../../../../../vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/IERC20.sol";
+import {ITypeAndVersion} from "../../../../../../shared/interfaces/ITypeAndVersion.sol";
+import {IRouterClient} from "../../../../../interfaces/IRouterClient.sol";
+import {IEVM2AnyOnRamp} from "../../../../../interfaces/IEVM2AnyOnRamp.sol";
+import {IAny2EVMOffRamp} from "../../../../../interfaces/IAny2EVMOffRamp.sol";
+import {IRouter as IRouterBase} from "../../../../../interfaces/IRouter.sol";
+import {Client} from "../../../../../libraries/Client.sol";
+import {Internal} from "../../../../../libraries/Internal.sol";
+import {UpgradeableLockReleaseTokenPool_Sepolia} from "./LegacyTestnetTokenPools/UpgradeableLockReleaseTokenPool_Sepolia.sol";
+import {UpgradeableBurnMintTokenPool_ArbSepolia} from "./LegacyTestnetTokenPools/UpgradeableBurnMintTokenPool_ArbSepolia.sol";
+
+interface IRouter is IRouterClient, IRouterBase {
+  struct OffRamp {
+    uint64 sourceChainSelector;
+    address offRamp;
+  }
+  function getWrappedNative() external view returns (address);
+  function isOffRamp(uint64, address) external view returns (bool);
+  function getOffRamps() external view returns (OffRamp[] memory);
+}
+
+interface IEVM2EVMOffRamp_1_2 is IAny2EVMOffRamp, ITypeAndVersion {
+  function executeSingleMessage(Internal.EVM2EVMMessage memory message, bytes[] memory offchainTokenData) external;
+}
+
+interface IEVM2EVMOffRamp_1_5 is IAny2EVMOffRamp, ITypeAndVersion {
+  function executeSingleMessage(
+    Internal.EVM2EVMMessage calldata message,
+    bytes[] calldata offchainTokenData,
+    uint32[] memory tokenGasOverrides
+  ) external;
+}
```

Comment:
no we cannot; they don't have the additional struct & methods in IRouter separately in an interface, and regarding off ramp we don't have 1.5 interface in this repo

Reply:
this is weird. probably an oversight [on their end](https://github.com/smartcontractkit/ccip/blob/ccip-develop/contracts/src/v0.8/ccip/Router.sol#L210)
can you raise an issue for that?

Reply:
i think it's intentional bc the IRouter interface on their end only exposes methods that other contracts would externally call; for offramp; `executeSingleMessage` is absent because it is onlyCallable by [self](https://github.com/aave/ccip/blob/65a5b9ceb2e0d4aef77550c0353ca841f8838995/contracts/src/v0.8/ccip/offRamp/EVM2EVMOffRamp.sol#L422), we use this method to avoid mocking the root gen & proof verification part

Reply:
makes sense